data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaProducerConfig.java

```diff
@@ -0,0 +1,28 @@
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+public interface KafkaProducerConfig extends KafkaConnection {
+    Collection<String> getBootStrapServers();
+
+    AuthConfig getAuthConfig();
+
+    SchemaConfig getSchemaConfig();
+
+    String getSerdeFormat();
+
+    List<TopicConfig> getTopics();
+
+    KafkaProducerProperties getKafkaProducerProperties();
+
+    void setDlqConfig(PluginSetting pluginSetting);
+
+    String getPartitionKey();
+
+    Optional<PluginModel> getDlq();
```

Comment:
DLQ should be sink specific and not producer specific.

Reply:
I don't think it requires substantial changes. We just have to include it in the kafka sink config and not in the producer config. You can do it in the next PR.